### PR TITLE
Add changelog entry for BoolList.

### PR DESCRIPTION
**Pull request by lrhn**
Add author to AUTHORS.
Small cleanups.
Rename `.from` constructor to `.of`, for consistency.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -4,3 +4,4 @@
 #   Name/Organization <email address>
 
 Google Inc.
+AAABramenko (https://github.com/AAAbramenko)
```
Comment (lrhn) [inline AUTHORS:7]: @AAAbramenko Do you want an entry in this file, or would you prefer not to? Do you want your e-mail listed here or just your github link?

Comment (AAAbramenko) [reply AUTHORS:7]: I'll be glad. I think nickname and github link are enough.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: collection
-version: 1.16.0
+version: 1.17.0-dev
```
Comment (natebosch) [inline pubspec.yaml:2]: We are still on `1.16.0`.

https://github.com/dart-lang/collection/pull/211

**Review by natebosch — APPROVED** [review]

File: pubspec.yaml (hunk 2 above)
Comment (lrhn) [reply pubspec.yaml:2]: ACK.

**Review by natebosch — APPROVED** [review]

**Pull request merged**